tv-shows/src/components/core/IconInput/IconInput.tsx

```diff
@@ -0,0 +1,58 @@
+'use client';
+
+import { Icon, Input, InputGroup, InputLeftAddon, InputRightElement } from '@chakra-ui/react';
+import { useEffect, useRef, useState } from 'react';
+import { FieldValues, Path, UseFormRegister } from 'react-hook-form';
+import { IconType } from 'react-icons';
+import { FaEye, FaEyeSlash } from 'react-icons/fa6';
+
+interface IIconInputProps<T extends FieldValues> {
+	icon: IconType;
+	placeholder: string;
+	type: string;
+	formControlName: Path<T>;
+	register: UseFormRegister<T>;
+}
+
+export function IconInput<T extends FieldValues>({
```

Comment:
Cool generic, but I'd rather just do:
`IconInputProps extends InputProps {}`
and then 
`<Input ... {...rest}>`

Reply:
Will fix, VSCode is playing tricks on me